File: JavaScript/string_examples.js
"use strict";
var s2;

// Make a string
var s = "Hello";

// Find length of a string
console.log(s.length);

// Compare two strings
s2 = "Hello 2";
console.log(s == s2);

// Search for a substring
s2 = "el";
console.log(s.indexOf(s2));

// Make a substring
var b = 1;
var e = 3;
s2 = s.substring(b, e);
console.log(s2);

// Split a string
s = "Hello how are you?";
console.log(s.split(" "));

File: C++/string_examples.cpp
#include <iostream>

int main(int argc, const char * argv[]) {

  // Make a string
  std::string s("Hello");

  // Find length of a string
  std::cout << s.length() << std::endl;

  // Compare two strings
  std::string s2("Hello 2");
  std::cout << (s == s2) << std::endl;

  // Search for a substring
  s2 = std::string("el");
  std::cout << s.find(s2) << std::endl;

  // Make a substring
  size_t b = 1;
  size_t e = 3;
  s2 = s.substr(b, e);
  std::cout << s2 << std::endl;

  return 0;
}
